### PR TITLE
Bump version to 21.2.1

### PR DESCRIPTION
**Pull request by leighmcculloch**
### What
Bump version to 21.2.1, creating release branch.

### Why

Created by @leighmcculloch manually because the Bump Version workflow is currently erroring with permission errors. Related links about those failures:
- https://github.com/stellar/rs-soroban-env/actions/runs/10446396038/job/28930412319
- https://stellarfoundation.slack.com/archives/C02U19A2A/p1723857725537729

### What is next

See the release instructions for a full rundown on the release process:
https://github.com/stellar/actions/blob/main/README-rust-release.md

Commit any changes to the `release/v21.2.1` branch that are needed in this release.

~~If this is a regular release releasing from `main`, merge this PR when ready, and after merging, create a release for this version by going to this link: https://github.com/stellar/rs-soroban-env/releases/new?tag=v21.2.1&title=21.2.1~~

If this is a backport or patch release of a past version, see the release instructions. When ready to release this branch create a release by going to this link: 
https://github.com/stellar/rs-soroban-env/releases/new?tag=v21.2.1&title=21.2.1&target=release/v21.2.1

### CODE REVIEW

File: .github/workflows/rust.yml [hunk 1]
```diff
@@ -6,6 +6,9 @@ on:
   pull_request:
   merge_group:
 
+env:
+  RUSTFLAGS: -Aunexpected_cfgs
```
Comment (leighmcculloch) [inline .github/workflows/rust.yml:10]: The `unexpected_cfgs` error was added in Rust 1.80, but this release is forking off a past release, v21.2.0, that doesn't compile when that error is enabled, for reasons that are not material.

On the `main` branch the things causing the error to occur will, or have been already, fixed. But for the sake of a patch release that has the goal of making a small change on a past release, silencing the error is more appropriate.

**Comment by dmkozh** [comment]
> [leighmcculloch](https://github.com/leighmcculloch) wants to merge 2 commits into [main](https://github.com/stellar/rs-soroban-env/tree/main)

This should go into a separate branch, not into main, right?

**Comment by leighmcculloch** [comment]
> > [leighmcculloch](https://github.com/leighmcculloch) wants to merge 2 commits into [main](https://github.com/stellar/rs-soroban-env/tree/main?rgh-link-date=2024-08-19T15%3A12%3A56Z)
> 
> This should go into a separate branch, not into main, right?

I'll close the PR without merging it once tagged/released. We don't need the change to go to any branch, it's just a visible place to track and open piece of work.

**Review by dmkozh — APPROVED** [review]

**Comment by leighmcculloch** [comment]
Released in https://github.com/stellar/rs-soroban-env/releases/tag/v21.2.1

**Pull request closed without merging**